_inc/lib/class.core-rest-api-endpoints.php

```diff
@@ -881,11 +881,22 @@ public static function set_connection_owner( $request ) {
 			);
 		}
 
+		// Update the master user in Jetpack
 		$updated = Jetpack_Options::update_option( 'master_user', $new_owner_id );
-		if ( $updated ) {
+
+		// Notify WPCOM about the master user change
+		Jetpack::load_xml_rpc_client();
+		$xml = new Jetpack_IXR_Client( array(
+			'user_id' => get_current_user_id(),
+		) );
+		$xml->query( 'jetpack.switchBlogOwner', array(
+			'new_blog_owner' => $new_owner_id,
+		) );
```

Comment:
One question here:

First, local site `master_user` option is updated.
Second, `new_blog_owner` update request is sent to WPCOM.

Nothing links these together and makes them change atomically, either both or neither. Especially the network-dependent bit updating WPCOM could fail. Should we be concerned about these things getting out of sync?

Reply:
I don't think we should be concerned about them going out of sync - if they do for some odd reason, it'll regardless need a HE intervention.

FWIW, when Jetpack supported this before (in the old admin, before Jetpack React), it worked the same way - see https://github.com/Automattic/jetpack/blob/280eae17dc7e92e6c4ab75403a331a9da5206062/_inc/lib/admin-pages/class.jetpack-my-jetpack-page.php#L42 and https://github.com/Automattic/jetpack/blob/280eae17dc7e92e6c4ab75403a331a9da5206062/_inc/lib/admin-pages/class.jetpack-my-jetpack-page.php#L50